contrib/autofs/patches/musl.patch

```diff
@@ -0,0 +1,39 @@
+fixes:
+```
+../include/hash.h:27:2: error: Wordsize not 32 or 64                                                                                                                 
+   27 | #error Wordsize not 32 or 64                                                                                                                                 
+      |  ^                                                                        
+```
+
+--- a/include/hash.h
++++ b/include/hash.h
+@@ -6,7 +6,7 @@
+ #ifdef __GLIBC__
+ #include <bits/wordsize.h>
+ #else
+-#include <bits/reg.h>
++#include <sys/reg.h>
+ #endif
+ #include <linux/stddef.h>
+ #include <sys/types.h>
+
+fixes:
+```
+automount.c:2541:22: error: call to undeclared function 'basename'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
+ 2541 |                         const char *map = basename(name);
+      |                                           ^
+automount.c:2541:16: error: incompatible integer to pointer conversion initializing 'const char *' with an expression of type 'int' [-Wint-conversion]
+ 2541 |                         const char *map = basename(name);
+      |                                     ^     ~~~~~~~~~~~~~~
+```
+
+--- a/include/automount.h
++++ b/include/automount.h
+@@ -20,6 +20,7 @@
+ #include <unistd.h>
+ #include <fcntl.h>
+ #include <mntent.h>
++#include <libgen.h>
```

Comment:
this is not correct, autofs uses basename that takes a constant string while the posix libgen.h version potentially changes its input

note the build warning:

```
/builddir/autofs-5.1.9/daemon/automount.c:2541:31: warning: passing 'const char *' to parameter of type 'char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
```